core_backend/add_new_data_to_db.py

```diff
@@ -0,0 +1,412 @@
+    a confirmation.
+    The feedback should also be tailored to the sentiment of the user's question and
+    should be short and concise especially when the feedback is positive.
+    Your output should strictly adhere to the following format
+
+    {{"output":"<feedback_text>"}}
+
+    Here are the inputs:
+    - Question: "{question_text}"
+    - FAQ: "{faq_text}"
+    - Sentiment: "{sentiment}"  // The sentiment can be "positive" or "negative"
+
+    Generate the feedback text based on the sentiment and input.
+
+    Output:
+    """
+
+    response = completion(
+        model=LITELLM_MODEL_DEFAULT,
+        api_base=LITELLM_ENDPOINT,
+        api_key=LITELLM_API_KEY,
+        messages=[{"role": "user", "content": prompt}],
+        max_tokens=100,
+        temperature=0.7,
+    )
+
+    # Extract the output from the response
+    feedback_output = response["choices"][0]["message"]["content"].strip()
+    feedback_output = feedback_output.replace("json", "")
+    feedback_output = feedback_output.replace("\n", "").strip()
+
+    try:
+        feedback_dict = json.loads(feedback_output)
+        if isinstance(feedback_dict, dict) and "output" in feedback_dict:
+
+            return feedback_dict
+        else:
+            raise ValueError("Output is not in the correct format.")
+    except (SyntaxError, ValueError) as e:
+        print(f"Output is not in the correct format.{e}")
+        return None
+
+
+def save_single_row(endpoint: str, data: dict) -> dict:
+    """
+    Save a single row in the database.
+    """
+
+    response = requests.post(
+        endpoint,
+        headers={
+            "accept": "application/json",
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {API_KEY}",
+        },
+        json=data,
+        verify=False,
+    )
+    return response.json()
+
+
+def process_search(_id: int, text: str) -> tuple | None:
+    """
+    Process and add query to DB
+    """
+
+    endpoint = f"{HOST}/search"
+    data = {
+        "query_text": text,
+        "session_id": 0,
+        "generate_llm_response": False,
+        "query_metadata": {"some_key": "some_value"},
+        "generate_tts": False,
+    }
+    response = save_single_row(endpoint, data)
+    if "search_results" in response:
+        return (
+            _id,
+            response["query_id"],
+            response["feedback_secret_key"],
+            response["search_results"],
+        )
+    return None
+
+
+def process_response_feedback(
+    query_id: int, feedback_sentiment: str, feedback_secret_key: str, is_off_topic: bool
+) -> tuple | None:
+    """
+    Process and add response feedback to DB
+    """
+
+    endpoint = f"{HOST}/response-feedback"
+
+    if is_off_topic and feedback_sentiment == "positive":
+        return None
+
+    data = {
+        "query_id": query_id,
+        "session_id": 0,
+        "feedback_sentiment": feedback_sentiment,
+        "feedback_text": "",
```

Comment:
We can keep this None or drop this field altogether also. Will store as null feeedback_text!